src/haz3lweb/Grading.re

```diff
@@ -265,12 +270,12 @@ module ImplGradingReport = {
   //   );
   // };
 
-  let individual_report = (i, ~inject, ~hint: string, ~status) =>
+  let individual_report = (i, ~inject, ~hint: string, ~status, (id, _)) =>
     div(
       ~attr=
         Attr.many([
           Attr.classes(["test-report"]),
-          Attr.on_click(TestView.jump_to_test(~inject)),
+          Attr.on_click(TestView.jump_to_test(~inject, -1, id)),
```

Comment:
ah I see, it was a hack to not have to compute the number of editors above. let's not do a hack like this -- you should define a helper function to compute the correct index

Reply:
Didn't change the -1 here because the module here doesn't have the access to the information necessary to determine the number of editors (this module only handles the ImplementationGradingReport, we need the number of WrongImplementation to determine the number of editors). So I postpone this responsibility to switch_editor function.

Reply:
If there is information that is needed here, add it as an argument to the function so the caller is responsible for providing that information. Might require threading it through a few levels, but better than using `-1`. If that really isn't feasible, you should use an option type rather than a special value.